### PR TITLE
[191] unit

### PR DESCRIPTION
**Pull request by piotr-s-brainhub**
Deployed to https://beghp.github.io/gh-pages-rc-3<br>It resolves https://github.com/brainhubeu/react-carousel/issues/191<br><br>- [ ] `package.json:version` has been updated with `npm version patch` (or `major/minor` as appropriate)<br>- [ ] `@brainhubeu/react-carousel` version has been updated in `docs-www/package.json` to the same which is in `package.json:version`<br><br>

### CODE REVIEW

**Comment by piotr-s-brainhub** [comment]
I validated, it works:
Screenshot 2020-02-23 at 01 01 49

After moving to the next slide, the slide can be hidden but it occurs without specifying the unit as well:
Screenshot 2020-02-23 at 01 03 13

File: src/components/CarouselItem.js [hunk 1]
```diff
@@ -35,11 +37,11 @@ class CarouselItem extends PureComponent {
           },
         )}
         style={{
-          paddingRight: `${this.props.offset / 2}px`,
-          paddingLeft: `${this.props.offset / 2}px`,
-          width: `${this.props.width}px`,
-          maxWidth: `${this.props.width}px`,
-          minWidth: `${this.props.width}px`,
+          paddingRight: `${this.props.offset / 2}${this.props.offsetUnit}`,
```
Comment (humbak) [inline src/components/CarouselItem.js]: I'm not sure how it works. If everything is calculated in pixels and you will switch just the unit that will multiply everything. I think you should calculate everything in rems. Did you test that solution?

File: src/components/Carousel.js [hunk 2]
```diff
@@ -589,7 +591,9 @@ class Carousel extends Component {
                 currentSlideIndex={this.getActiveSlideIndex()}
                 index={index}
                 width={this.getCarouselElementWidth()}
+                widthUnit={this.props.itemWidthUnit || 'px'}
                 offset={index !== slides.length ? this.props.offset : 0}
+                offsetUnit={this.props.offsetUnit || 'px'}
```
Comment (humbak) [inline src/components/Carousel.js]: please use defaultProps instead

File: README.md [hunk 3]
```diff
@@ -215,7 +215,9 @@ Where:
 * `autoPlay` (*number*) slide change interval in milliseconds
 * `stopAutoPlayOnHover` (*boolean*) determines if autoPlay should stop when mouse hover over carousel
 * `offset` (*number*) padding between items
+* `offsetUnit` (*string*, default `px`) - CSS unit of `offset`
 * `itemWidth` (*number*) determines custom width for each slide in carousel
+* `itemWidthUnit` (*string*, default `px`) - CSS unit of `itemWidth`
```
[the remark this answers is not in the document]
Comment (piotr-s-brainhub) [reply README.md]: fixed in 5729d2c260be3720be52d568a89d239b39efaa71

File: src/components/Carousel.js [hunk 4]
```diff
@@ -589,7 +591,9 @@ class Carousel extends Component {
                 currentSlideIndex={this.getActiveSlideIndex()}
                 index={index}
                 width={this.getCarouselElementWidth()}
+                widthUnit={this.props.itemWidthUnit || 'px'}
```
[the remark this answers is not in the document]
Comment (piotr-s-brainhub) [reply src/components/Carousel.js]: fixed in be70f5a98c2ea1a97573d027e8d0c03c394d527f

File: src/components/Carousel.js (hunk 2 above)
Comment (piotr-s-brainhub) [reply src/components/Carousel.js]: fixed in be70f5a98c2ea1a97573d027e8d0c03c394d527f

File: src/components/CarouselItem.js [hunk 5]
```diff
@@ -10,7 +10,9 @@ class CarouselItem extends PureComponent {
     clickable: PropTypes.bool,
     children: PropTypes.node,
     width: PropTypes.number,
+    widthUnit: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (piotr-s-brainhub) [reply src/components/CarouselItem.js]: fixed in 0e0e3055a3059c7e7cc5ce73cc941dbf68e50118

File: src/components/CarouselItem.js [hunk 6]
```diff
@@ -10,7 +10,9 @@ class CarouselItem extends PureComponent {
...
     offset: PropTypes.number,
+    offsetUnit: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (piotr-s-brainhub) [reply src/components/CarouselItem.js]: fixed in 0e0e3055a3059c7e7cc5ce73cc941dbf68e50118

File: src/components/CarouselItem.js (hunk 1 above)
Comment (piotr-s-brainhub) [reply src/components/CarouselItem.js]: I checked sometimes `%` works even better than `px`, `em` or `rem`
https://recordit.co/5dpI9Adpcv

**Pull request closed without merging**